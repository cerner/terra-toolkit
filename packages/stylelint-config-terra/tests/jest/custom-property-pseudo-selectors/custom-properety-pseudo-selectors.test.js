/**
 * @jest-environment node
 */
const stylelint = require('stylelint');
const path = require('path');

const config = {
  plugins: [
    path.resolve(__dirname, '..', '..', '..', 'src', 'rules', 'custom-property-pseudo-selectors', 'custom-property-pseudo-selectors'),
  ],
  rules: {
    'terra/custom-property-pseudo-selectors': true,
  },
};

describe('custom-property-pseudo-selectors', () => {
  it('does not error with a longer prefix before the pseudo selector', () => {
    const result = stylelint.lint({
      code: '.test:hover { color: var(--test-example-hover-color) }',
      config,
    });

    return result.then(data => expect(data.errored).toBeFalsy());
  });

  it('does not error with a single pseudo selector', () => {
    const result = stylelint.lint({
      code: '.test:focus { color: var(--test-focus-color) }',
      config,
    });

    return result.then(data => expect(data.errored).toBeFalsy());
  });

  it('does not error with a double pseudo selector', () => {
    const result = stylelint.lint({
      code: '.test:focus:hover { color: var(--test-focus-hover-color) }',
      config,
    });

    return result.then(data => expect(data.errored).toBeFalsy());
  });

  it('does not error with a compound pseudo selector', () => {
    const result = stylelint.lint({
      code: '.one:focus.two:hover { color: var(--test-focus-hover-color) }',
      config,
    });

    return result.then(data => expect(data.errored).toBeFalsy());
  });

  it('does not error with two nested pseudo selectors', () => {
    const result = stylelint.lint({
      code: '.one:focus { .two:hover { color: var(--test-one-focus-two-hover-color) } }',
      config,
    });

    return result.then(data => expect(data.errored).toBeFalsy());
  });

  it('does not error with three nested pseudo selectors', () => {
    const result = stylelint.lint({
      code: '.one:focus { .two:hover, .three:hover { color: var(--test-one-focus-two-hover-color) } }',
      config,
    });

    return result.then(data => expect(data.errored).toBeFalsy());
  });

  it('does error with a custom properties that does not include the pseudo selector', () => {
    const result = stylelint.lint({
      code: '.test:hover { color: var(--test-color) }',
      config,
    });

    return result.then(data => {
      expect(data.errored).toBeTruthy();
      expect(data.results[0].warnings).toMatchSnapshot();
    });
  });

  it('does error with a custom properties that does not include the ancestor pseudo selectors', () => {
    const result = stylelint.lint({
      code: '.test:hover { color: var(--test-color) }',
      config,
    });

    return result.then(data => {
      expect(data.errored).toBeTruthy();
      expect(data.results[0].warnings).toMatchSnapshot();
    });
  });

  it('does error with a custom properties that declares pseudo selectors in the wrong order', () => {
    const result = stylelint.lint({
      code: '.one:hover { .two:focus { color: var(--test-focus-hover-color) } }',
      config,
    });

    return result.then(data => {
      expect(data.errored).toBeTruthy();
      expect(data.results[0].warnings).toMatchSnapshot();
    });
  });
});
