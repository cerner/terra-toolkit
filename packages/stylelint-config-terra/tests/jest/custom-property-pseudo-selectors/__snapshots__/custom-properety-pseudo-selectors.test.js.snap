// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom-property-pseudo-selectors does error with a custom properties that declares pseudo selectors in the wrong order 1`] = `
Array [
  Object {
    "column": 27,
    "line": 1,
    "rule": "terra/custom-property-pseudo-selectors",
    "severity": "error",
    "text": "Expected --test-focus-hover-color to include all ancestor psuedo selectors in order: [hover,focus] (terra/custom-property-pseudo-selectors)",
  },
]
`;

exports[`custom-property-pseudo-selectors does error with a custom properties that does not include the ancestor pseudo selectors 1`] = `
Array [
  Object {
    "column": 15,
    "line": 1,
    "rule": "terra/custom-property-pseudo-selectors",
    "severity": "error",
    "text": "Expected --test-color to include all ancestor psuedo selectors in order: [hover] (terra/custom-property-pseudo-selectors)",
  },
]
`;

exports[`custom-property-pseudo-selectors does error with a custom properties that does not include the pseudo selector 1`] = `
Array [
  Object {
    "column": 15,
    "line": 1,
    "rule": "terra/custom-property-pseudo-selectors",
    "severity": "error",
    "text": "Expected --test-color to include all ancestor psuedo selectors in order: [hover] (terra/custom-property-pseudo-selectors)",
  },
]
`;
